gematria/basic_block/basic_block.h

```diff
@@ -218,6 +218,32 @@ class InstructionOperand {
 
 std::ostream& operator<<(std::ostream& os, const InstructionOperand& operand);
 
+// Represents an annotation holding a value such as some measure/statistic
+// paired with the instruction.
+struct Annotation {
+  // Values default to -1 to have name only annotations.
+  Annotation() : value(-1){};
```

Comment:
It's a bit more complex, but maybe we could use a [NaN](https://en.cppreference.com/w/cpp/numeric/math/NAN) for name-only annotations? That way, we could have any valid numerical value as the annotation value.

Reply:
I was wondering whether name only annotations should be a thing - I wrote them in for the `proto/**` and `basic_block/**` sides of things, but didn't add support for them from the model's side of things, since that would fit in with the current tensor-overwriting implementation.